mantidproject/mantid
# Use py3compat Enums in ISIS SANS

DO NOT MERGE INTO MASTER UNTIL RELEASE BRANCH HAS BEEN CREATED

**Description of work.**
ISIS SANS had implemented pseudo-enums in python. This PR refactors SANS to use the `Enum` class from `mantid.py3compat`

In most places I have used the oneline init syntax for enums:
```
Foo = Enum("Foo", "foo bar")
```
but in one file I decided to use 
```
class Foo(Enum):
    foo = 1
    bar = 2
```
because one of the enums had many values, so the first syntax would not be clear. Enum init style has been kept constant at the file-level
There is no difference between these syntax.

**To test:**
Code review: Primarily looking to check if I've misspelled any of the enum values, or for dodgy regex when removing `to_string` and `from_string` method usage

* Interfaces -> SANS -> ISIS SANS
* Play around with the buttons at the bottom of the gui e.g. output mode, reduction dimension, file types
* Load the attached user and batch file
* Click `Process All`
* Processing should be successful (row turns green)

Fixes #25358 

*This does not require release notes* because **internal code change**

**Data:**
[SANS2DprData.zip](https://github.com/mantidproject/mantid/files/3334921/SANS2DprData.zip)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
